src/server/pkg/worker/api_server.go

```diff
@@ -224,6 +227,9 @@ func (a *APIServer) downloadData(logger *taggedLogger, inputs []*Input, puller *
 
 // Run user code and return the combined output of stdout and stderr.
 func (a *APIServer) runUserCode(ctx context.Context, logger *taggedLogger, environ []string) error {
+	defer func(start time.Time) {
+		logger.Logf("running user code for environment (%v) took %v\n", environ, time.Since(start))
+	}(time.Now())
```

Comment:
We already have a log statement for finishing user code execution, we should probably delete the other one since this has more information. I'm not sure there's much value in logging the environment though, that doesn't really change datum to datum.